### PR TITLE
Completed test coverage kernellist read config

### PR DESCRIPTION
**Pull request by adrianseco**
## 🗒️ Summary
Added unit tests  to achieve the full coverage to read_config method in KernelList class.

## ⚙️ Test Data and/or Report
Added 3 new unit tests to check read_config method to achieve the full coverage.

## ♻️ Related Issues
Fixes #195

### CODE REVIEW

File: tests/npb/test_classes_kernel_list.py [hunk 1]
```diff
@@ -76,4 +83,161 @@ def test__init__(self, tmp_path, pds_version, dataset_i,
...
+        # Check that the regular expressions work.
+        assert kernel_list.re_config[0].match('maven_001.bsp')
+        assert not kernel_list.re_config[0].match('maven_001.tsc')
+        assert kernel_list.re_config[1].match('maven_release_03.tm')
+        assert not kernel_list.re_config[1].match('release_03.tm')
+
+        # Check that read_config has correctly generated the formatted
+        # representation of the JSON.
+        assert (kernel_list.json_formatted_lst ==
+                json.dumps(json_config, indent=2).split('\n'))
```
Comment (jdiazdelrio) [inline tests/npb/test_classes_kernel_list.py]: This fragment
```
json.dumps(json_config, indent=2).split('\n')
```
is the same as the sequence of call implemented by the code itself. Instead, provide a literal representation.

File: tests/npb/test_classes_kernel_list.py [hunk 2]
```diff
@@ -76,4 +83,161 @@ def test__init__(self, tmp_path, pds_version, dataset_i,
...
+
+        # Also exposes the same compiled regex list through setup. This is
+        # important because other code consumes setup.re_config later.
+        assert kernel_list.setup.re_config is kernel_list.re_config
+
+        # The method read_config must create the pretty-printed JSON
+        # representation used later when logging or displaying the kernel list
+        # configuration.
+        assert (kernel_list.json_formatted_lst == json.dumps(json_config,
+                                                             indent=2).split('\n'))
```
Comment (jdiazdelrio) [inline tests/npb/test_classes_kernel_list.py]: Same comment as for `test__init__calls_real_read_config` case. Provide the pretty-printed JSON representation so that you don't repeat the code fragment used within the code.

**Review by jdiazdelrio — CHANGES_REQUESTED** [review]
The firs of the last two tests "exposes" a  corner case for which the code has no solution (it will actually crash). I would add a note to either the tests and/or the code indicating the issue.

In the case of `test_read_config_raises_attribute_error_when_setup_has_no_config`, I'm not sure it is possible because the verification of the `Setup` class would fail. Is it possible to have a configuration file without `kernel_list`, or a `kernel_list` block within the configuration file without a kernel?

**Review by jdiazdelrio — APPROVED** [review]
Thanks for this PR. Let's keep in mind the two notes in the tests when we update the code.
